### PR TITLE
Field player avatar

### PR DESCRIPTION
**Pull request by PikalaxALT**
Might conflict with #310

### CODE REVIEW

File: src/new_menu_helpers.c [hunk 1]
```diff
@@ -496,7 +496,7 @@ void ClearDialogWindowAndFrame(u8 windowId, bool8 copyToVram)
     if (copyToVram == TRUE)
         CopyWindowToVram(windowId, 3);
     if (gQuestLogState == 2)
```
Comment (jiangzhengwenjz) [inline src/new_menu_helpers.c]: `QL_STATE_2`

File: src/metatile_behavior.c [hunk 2]
```diff
@@ -191,7 +191,7 @@ bool8 MetatileBehavior_IsLadder(u8 metatileBehavior)
         return FALSE;
 }
 
-bool8 MetatileBehavior_IsCaveDoor(u8 metatileBehavior)
+bool8 MetatileBehavior_IsNonAnimDoor(u8 metatileBehavior)
 {
     if(metatileBehavior == MB_CAVE_DOOR)
```
Comment (jiangzhengwenjz) [inline src/metatile_behavior.c]: so cave door is the only non-anim door?

File: src/field_player_avatar.c [hunk 3]
```diff
@@ -0,0 +1,2165 @@
...
+    }
+
+    return FALSE;
+}
+
+static void npc_clear_strange_bits(struct ObjectEvent *objEvent)
+{
+    objEvent->inanimate = 0;
+    objEvent->disableAnim = 0;
+    objEvent->facingDirectionLocked = 0;
```
Comment (jiangzhengwenjz) [inline src/field_player_avatar.c]: `FALSE`

File: src/metatile_behavior.c (hunk 2 above)
Comment (PikalaxALT) [reply src/metatile_behavior.c]: this was a rename to sync with emerald. do you prefer the previous name?

File: src/event_object_movement.c [hunk 4]
```diff
@@ -153,7 +153,7 @@ static void sub_80632BC(struct Sprite *);
 static void sub_80632E0(struct Sprite *);
 static void MovementType_WanderAroundDuplicate(struct Sprite *);
 
-extern void sub_805B3B8(struct Sprite *);
+extern void MovementType_Player(struct Sprite *);
```
Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: should be in `include/field_player_avatar.h`

File: src/event_object_movement.c [hunk 5]
```diff
@@ -4917,6 +4917,8 @@ static void ClearObjectEventMovement(struct ObjectEvent *objectEvent, struct Spr
     sprite->data[1] = 0;
 }
 
+// Off by one error
```
Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: Can you resolve them?

**Pull request merged**